python/tests/server/test_worker.py

```diff
@@ -475,57 +510,64 @@ def setup(self):
         try:
             self.setup_generator = self.worker.setup()
             self.setup_events = []
+            # read one event so that we raise InvalidStateException
+            event = self.await_(anext(self.setup_generator))
+            self.setup_events.append(event)
         except InvalidStateException:
-            pass
+            self.setup_generator = None
 
     @precondition(lambda x: x.setup_generator)
     @rule(n=st.integers(min_value=1, max_value=10))
     def read_setup_events(self, n):
         try:
             for _ in range(n):
-                event = next(self.setup_generator)
+                event = self.await_(anext(self.setup_generator))
                 self.setup_events.append(event)
-        except StopIteration:
+        except StopAsyncIteration:
             self.setup_generator = None
+            self.await_(self._check_setup_events())
+        # except InvalidStateException:
+        #     pass
 
-            self._check_setup_events()
-
-    def _check_setup_events(self):
+    async def _check_setup_events(self):
```

Comment:
if you leave a non-async `_process` available to yourself (as suggested above) you don't need to do this either.